### ISSUE TITLE
Add `valid-selector` rule

### ISSUE THREAD

**Issue by DetachHead** [playwright-community/eslint-plugin-playwright#125]
```ts
page.click('div[foo=bar') // syntax error
```

**Comment by mskelton** [comment]
I'm not sure I like this rule, I think it would be very challenging to maintain especially as there are new features added to Playwright that can change the syntax for selectors, so this could easily result in a lot of false negatives as new versions are published.

**Comment by mxschmitt** [comment]
Especially in a world where we discourage page.* and .locator(*) methods. -> role based selectors are the recommendation nowadays.

**Comment by DetachHead** [comment]
> I'm not sure I like this rule, I think it would be very challenging to maintain especially as there are new features added to Playwright that can change the syntax for selectors, so this could easily result in a lot of false negatives as new versions are published.

i'm sure it could just call the same code playwright does to validate selectors. looks like that's `Selectors.parseSelector` in `playwright-core/lib/server/selectors.js`. that way there's no maintenence required

> Especially in a world where we discourage page.* and .locator(*) methods. -> role based selectors are the recommendation nowadays.

i find there are still plenty of cases where role based selectors aren't sufficient. the sites i use playwright on usually have really messy generated html so selectors are rarely simple enough to be covered by those methods.

**Comment by mskelton** [comment]
I'd be more comfortable if we use a Playwright API, that said I don't love the idea of using private methods from the Playwright source, that's pretty brittle unless Playwright exposes it officially which seems unlikely and unnecessary.

**Comment by mskelton** [comment]
I really don't like the idea of depending on internal Playwright source code, so I'm going to close this as not planned.
